packages/prover/src/proof_provider/proof_provider.ts

```diff
@@ -0,0 +1,200 @@
+import {Api, getClient} from "@lodestar/api/beacon";
+import {ChainForkConfig, createChainForkConfig} from "@lodestar/config";
+import {networksChainConfig} from "@lodestar/config/networks";
+import {Lightclient, LightclientEvent, RunStatusCode} from "@lodestar/light-client";
+import {LightClientRestTransport} from "@lodestar/light-client/transport";
+import {computeSyncPeriodAtSlot} from "@lodestar/light-client/utils";
+import {isForkExecution} from "@lodestar/params";
+import {allForks, capella} from "@lodestar/types";
+import {MAX_PAYLOAD_HISTORY, MAX_REQUEST_LIGHT_CLIENT_UPDATES} from "../constants.js";
+import {RootProviderOptions as RootProviderInitOptions} from "../interfaces.js";
+import {assertLightClient} from "../utils/assertion.js";
+import {getExecutionPayloads, getGenesisData, getSyncCheckpoint} from "../utils/consensus.js";
+import {numberToHex} from "../utils/conversion.js";
+import {OrderedMap} from "./ordered_map.js";
+
+interface RootProviderOptions extends RootProviderInitOptions {
+  transport: LightClientRestTransport;
+  api: Api;
+  config: ChainForkConfig;
+}
+
+export class ProofProvider {
+  private payloads: OrderedMap<string, allForks.ExecutionPayload> = new OrderedMap();
+  private finalizedPayloads: OrderedMap<string, allForks.ExecutionPayload> = new OrderedMap();
+  private readyPromise: Promise<void>;
+
+  // Map to match CL slot to EL block number
+  private slotsMap: {[slot: number]: string} = {};
+
+  lightClient?: Lightclient;
```

Comment:
wouldn't make it optional but assign in constructor

Reply:
Can't do that because of the comment below. https://github.com/ChainSafe/lodestar/pull/5222#discussion_r1134191019